Final take, not including bonus

deeprave opened the pull request:
Not entirely sure how to convert the detail views, but will take a
shot at it later and add it to this PR when working.

deeprave committed:
First take, with main entry points converted to model views.
Not entirely sure how to convert the detail views, but will have a
shot at it and add it when working.

MarkusH commented on timetracker/entries/forms.py:
```diff
@@ -1,23 +1,25 @@
 
 
-class EntryForm(forms.Form):
-    start = forms.DateTimeField(label="Start Time", help_text="Format: 2006-10-25 14:30")
```
You're loosing the help text for the form fields here. Either define them as `help_texts` on `EntryForm.Meta` or as part of the respective model field.

deeprave replied:
Whoops, fixed in coming update. Thanks!

MarkusH commented on timetracker/entries/urls.py:
```diff
@@ -4,9 +4,10 @@
 
 urlpatterns = [
-    url(r'^clients/$', views.clients, name='client-list'),
+	url(r'^$', views.RootRedirectView.as_view(), name='root'),
```
You should be careful with using tabs and spaces for indentation. Recommendation is to use 4 spaces.

deeprave replied:
Some weirdness with sublime text going on here. I caught it doing this once but missed this one. It is set to edit python code with no hard tabs. (I normally use PyCharm which to my knowledge never gets this wrong)

MarkusH commented on timetracker/entries/views.py:
```diff
@@ -1,31 +1,23 @@
-        'form': form,
-    })
+    def get_success_url(self):
+        return reverse('client-list')
```
You can use `success_url = reverse_lazy('client-list')` instead :wink: (`reverse_lazy` is at the same place as `reverse`)

MarkusH commented on timetracker/entries/views.py:
```diff
@@ -111,3 +88,10 @@ def project_detail(request, pk):
+class RootRedirectView(RedirectView):
+    permanent = False
+
+    def get_redirect_url(self):
```
You can use `redirect_url = reverse_lazy('client-list')` instead :wink: (`reverse_lazy` is at the same place as `reverse`)

MarkusH commented:
Looks good so far :smile:

deeprave committed:
Convert detail views to class views (based on UpdateView).
Fixed entries.html to use form snippet.
Use success_url/reverse_lazy as suggested by Markus.
